spec/models/custom_attribute_spec.rb

```diff
@@ -0,0 +1,13 @@
+describe CustomAttribute do
+  it "return correct field type" do
+    string_custom_attribute = FactoryGirl.create(:custom_attribute, :name => "foo", :value => "string", :resource_type => 'ExtManagementSystem')
```

Comment:
@imtayadeway ill split it up to 3 tests, didn't quite get the second comment if you could elaborate will be great.